lib/websocket/websocket.js

```diff
@@ -57,38 +58,48 @@ class WebSocket extends EventTarget {
     url = webidl.converters.USVString(url)
     protocols = options.protocols
 
-    // 1. Let urlRecord be the result of applying the URL parser to url.
+    // 1. Let baseURL be this's relevant settings object's API base URL.
+    const baseURL = getGlobalOrigin()
+
+    // 1. Let urlRecord be the result of applying the URL parser to url with baseURL.
     let urlRecord
 
     try {
-      urlRecord = new URL(url)
+      urlRecord = new URL(url, baseURL)
     } catch (e) {
-      // 2. If urlRecord is failure, then throw a "SyntaxError" DOMException.
+      // 3. If urlRecord is failure, then throw a "SyntaxError" DOMException.
       throw new DOMException(e, 'SyntaxError')
     }
 
-    // 3. If urlRecord’s scheme is not "ws" or "wss", then throw a
-    //    "SyntaxError" DOMException.
+    // 4. If urlRecord’s scheme is "http", then set urlRecord’s scheme to "ws".
+    if (urlRecord.protocol === 'http:') {
```

Comment:
Can you store the protocol in a variable so it doesn't trigger the getter everytime?

Reply:
Nit: Line 75 and 77 triggers the protocol getters, which can just be removed into one.